### PR TITLE
Added Babel dependency and configuration for building from ES6 codebase

### PR DESCRIPTION
**Pull request by gesposito**
I've made some config changes to let build from ES6 codebases as this one:
https://github.com/gesposito/gatsby-starter-blog

It's still hybrid (coffee, ES6) and just a proof of concept, but I'm willing to port it to a full ES6 starter.

### CODE REVIEW

File: lib/loaders/config-loader/index.js [hunk 1]
```diff
@@ -23,7 +23,6 @@
     value.relativePath = path.relative('.', directory);
     return globPages(directory, function(err, pagesData) {
       value.pages = pagesData;
-      this.value = [value];
```
Comment (KyleAMathews) [inline lib/loaders/config-loader/index.js]: Was this line not needed?

File: package.json [hunk 2]
```diff
@@ -73,5 +73,8 @@
   },
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "devDependencies": {
```
Comment (KyleAMathews) [inline package.json]: This should be a regular dependency at least for how I think about things. Gatsby *only* has a dev mode so what would normally be a dev dependency (like Babel) is just a regular dependency in Gatsby :)

**Comment by KyleAMathews** [comment]
Awesome! Made a few inline comments. LGTM.

P.s. if you want to contribute more and would prefer to do so in ES6, feel free to convert `.coffee` files over to es6 in the process. That's [the plan anyways](https://github.com/gatsbyjs/gatsby/issues/1#issuecomment-124858693), just really busy right now with work.

**Comment by KyleAMathews** [comment]
Also, if you just want to port the starters to es6, that'd be cool too. That should be the default. I can fork the original version as an explicit Coffeescript starter.

File: lib/loaders/config-loader/index.js (hunk 1 above)
Comment (gesposito) [inline lib/loaders/config-loader/index.js]: Yeah, forgot I bumped my head over it a while, might be some transformation that fails.
I'm not used to CoffeeScript now understand that @ notation is the this scope notation (right?), but while executing with that line this error pops up:
`Cannot set property 'value' of undefined`

For what I see the "value" is just passed in "callback" (wrapped as an array?).

File: package.json (hunk 2 above)
Comment (gesposito) [inline package.json]: Sure, just used to --save-dev babel :)

**Comment by gesposito** [comment]
I'm willing to port the blog starter, if I get used to the quirks of porting might port the rest too.
Not ready for prime time yet, will let you know when the blog one is done.

**Comment by gesposito** [comment]
I ended up converting the whole blog starter 
https://github.com/gesposito/gatsby-starter-blog

**Comment by KyleAMathews** [comment]
Awesome! I'd love a PR for that to the starter blog repo. Will give more feedback later to this PR.

**Comment by KyleAMathews** [comment]
Thanks! Merged and published https://github.com/gatsbyjs/gatsby/releases/tag/0.3.0

**Comment by gesposito** [comment]
Awesome!

**Comment by jlengstorf** [comment]
Hiya @gesposito! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Pull request merged**